### ISSUE TITLE
Trouble building example code

### ISSUE THREAD

**Issue by ChrisRovers** [faiface/pixel#122]
I'm going through the Building Pixel on Windows instructions and am stuck on actually doing go get in the platformer..

```
crove@DESKTOP-DD9HI9B MINGW64 /c/Users/crove/go/src/github.com/faiface/pixel/examples/platformer
$ go get
# github.com/faiface/pixel/examples/platformer
C:\Go\pkg\tool\windows_amd64\link.exe: running gcc failed: exit status 1
C:\Users\crove\AppData\Local\Temp\go-link-885384351\000002.o: In function `glfwSwapBuffers':
C:\Users\crove\go\src\github.com\go-gl\glfw\v3.2\glfw/glfw/src/context.c:609: undefined reference to `__assert_func'
C:\Users\crove\AppData\Local\Temp\go-link-885384351\000002.o: In function `glfwExtensionSupported':
C:\Users\crove\go\src\github.com\go-gl\glfw\v3.2\glfw/glfw/src/context.c:642: undefined reference to `__assert_func'
C:\Users\crove\AppData\Local\Temp\go-link-885384351\000002.o: In function `glfwGetProcAddress':
C:\Users\crove\go\src\github.com\go-gl\glfw\v3.2\glfw/glfw/src/context.c:707: undefined reference to `__assert_func'
C:\Users\crove\AppData\Local\Temp\go-link-885384351\000002.o: In function `glfwGetInputMode':
C:\Users\crove\go\src\github.com\go-gl\glfw\v3.2\glfw/glfw/src/input.c:153: undefined reference to `__assert_func'
C:\Users\crove\AppData\Local\Temp\go-link-885384351\000002.o: In function `glfwSetInputMode':
C:\Users\crove\go\src\github.com\go-gl\glfw\v3.2\glfw/glfw/src/input.c:174: undefined reference to `__assert_func'
C:\Users\crove\AppData\Local\Temp\go-link-885384351\000002.o:C:\Users\crove\go\src\github.com\go-gl\glfw\v3.2\glfw/glfw/src/input.c:262: more undefined references to `__assert_func' follow
collect2.exe: error: ld returned 1 exit status


crove@DESKTOP-DD9HI9B MINGW64 /c/Users/crove/go/src/github.com/faiface/pixel/examples/platformer
```

Looking around, I see a few suggestions in the go-gl issues tracker, but the issues were closed without resolution. I'm assuming it is a library version mismatch or something similar, but I installed versions as per the instructions (though I assume the pacman commands mostly just pulls the latest, which may have issues?)

```
$ gcc -v
Using built-in specs.
COLLECT_GCC=C:\msys64\mingw64\bin\gcc.exe
COLLECT_LTO_WRAPPER=C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/7.3.0/lto-wrapper.exe
Target: x86_64-w64-mingw32
Configured with: ../gcc-7.3.0/configure --prefix=/mingw64 --with-local-prefix=/mingw64/local --build=x86_64-w64-mingw32 --host=x86_64-w64-mingw32 --target=x86_64-w64-mingw32 --with-native-system-header-dir=/mingw64/x86_64-w64-mingw32/include --libexecdir=/mingw64/lib --enable-bootstrap --with-arch=x86-64 --with-tune=generic --enable-languages=c,lto,c++,objc,obj-c++,fortran,ada --enable-shared --enable-static --enable-libatomic --enable-threads=posix --enable-graphite --enable-fully-dynamic-string --enable-libstdcxx-time=yes --enable-libstdcxx-filesystem-ts=yes --disable-libstdcxx-pch --disable-libstdcxx-debug --disable-isl-version-check --enable-lto --enable-libgomp --disable-multilib --enable-checking=release --disable-rpath --disable-win32-registry --disable-nls --disable-werror --disable-symvers --with-libiconv --with-system-zlib --with-gmp=/mingw64 --with-mpfr=/mingw64 --with-mpc=/mingw64 --with-isl=/mingw64 --with-pkgversion='Rev2, Built by MSYS2 project' --with-bugurl=https://sourceforge.net/projects/msys2 --with-gnu-as --with-gnu-ld
Thread model: posix
gcc version 7.3.0 (Rev2, Built by MSYS2 project)

crove@DESKTOP-DD9HI9B MINGW64 /c/Users/crove/go/src/github.com/faiface/pixel/examples/platformer
```

Any suggestions?

**Comment by thegtproject** [comment]
Run through the msys part of this guide: https://thegtproject.github.io/post/preview-of-pixel-shaders/ and let me know if that fixes it

**Comment by NaniteFactory** [comment]
Today I came across the same problem.

It seems that there is an issue with https://github.com/go-gl/glfw/commit/46a8d530c3268b31eeb1310f30fe42be1cae98f2 which is currently the latest commit of `go-gl/glfw`.

I found that checkout to an older version does fix it. (For example, https://github.com/go-gl/glfw/commit/513e4f2bf85c31fba0fc4907abd7895242ccbe50)

Simply try:
```
cd $GOPATH/src/github.com/go-gl/glfw
git checkout 513e4f2
```

Hope this helps.
